x-pack/legacy/plugins/infra/server/routes/snapshot/types.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { InfraWrappableRequest } from '../../lib/adapters/framework';
+
+export interface InfraSnapshotNode {
+  path: InfraSnapshotNodePath[];
+
+  metric: InfraSnapshotNodeMetric;
+}
+
+export interface InfraSnapshotNodePath {
+  value: string;
+
+  label: string;
+
+  ip?: string | null;
+}
+
+export interface InfraSnapshotNodeMetric {
+  name: InfraSnapshotMetricType;
+
+  value?: number | null;
+
+  avg?: number | null;
+
+  max?: number | null;
+}
+
+export enum InfraSnapshotMetricType {
+  count = 'count',
+  cpu = 'cpu',
+  load = 'load',
+  memory = 'memory',
+  tx = 'tx',
+  rx = 'rx',
+  logRate = 'logRate',
+}
+
+export interface InfraNodeIdsInput {
+  nodeId: string;
+
+  cloudId?: string | null;
+}
+
+export enum InfraNodeType {
+  pod = 'pod',
+  container = 'container',
+  host = 'host',
+}
+
+export interface InfraTimerangeInput {
+  /** The interval string to use for last bucket. The format is '{value}{unit}'. For example '5m' would return the metrics for the last 5 minutes of the timespan. */
+  interval: string;
+  /** The end of the timerange */
+  to: number;
+  /** The beginning of the timerange */
+  from: number;
+}
+
+export enum InfraMetric {
```

Comment:
Same as above... Couldn't we use `InventoryMetricRT` and `InventoryMetric`  respectively from `common/inventory_models/types`?